src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -1501,18 +1495,20 @@ protected static String boolToStr(boolean value)
         return value ? "1" : "0";
     }
 
-    /**
-     * @deprecated  We believe that this method is no longer called by any part of the codebase, and it will soon be deleted.
-     */
-    @Deprecated
-    public static void deleteAllRecordsInGeneticProfile(long geneticProfileId) throws DaoException {
+    public static void deleteAllRecordsInGeneticProfileForSample(long geneticProfileId, long internalSampleId) throws DaoException {
```

Comment:
I haven't found any code using `public static void deleteAllRecordsInGeneticProfile(long geneticProfileId) throws DaoException`.

Removing mutations when the study is being removed happens due to the chain of cascade deleting foreign keys, right?

Reply:
My memory is that there are two separate DaoCancerStudy deleteCancerStudy functions ... one which relies on the cascade of foreign keys and one which explicitly deletes all data types with a separate dao delete function call for each data type / involved table. But if there are no calls to this function in the cbioportal-core codebase then I think deleting it is fine. (I don't think it will break the MSK importer, but if so, we can work around it)